### DISCUSSION TITLE
Setting up a temperature/humidity monitor but not sure where to start

### DISCUSSION THREAD

**Discussion by cookin83** [micropython/micropython#9394] [Core Development]
I have a Raspberry Pi Pico W and a sht30 temperature/humidity sensor that I would like to log data and publish a graph to a website to monitor my honey bee hive over the winter. I would like to save the data to a flash drive on a Raspberry Pi 3 so I can compare different months and years. Would it be easiest to set up a SQL database on the Pi? (I know very little about SQL) If so how do I log the data from the Pico to the Pi? Can a SQL database be converted into a line graph? Sorry if there is an obvious solution for this that I'm not finding. I think my knowledge of the Pico and SQL are so limited I'm not searching the right questions to get what I need. I was able to combine the Raspberry Pi get started guide for the Pico W and this forum thread [https://forum.micropython.org/viewtopic.php?f=21&t=12900&start=30](url) to get the live temperature and humidity to display on a local web page but I'm not sure where to go from here.

**Comment by DavesCodeMusings** [comment]
Unless you're dead set on DIY, you can probably achieve what you want using a [Home Assistant](https://www.home-assistant.io/) / [ESPHOME](https://esphome.io/) off-the-shelf solution. The Pi 3 would be your Home Assistant setup. It would collect the data and store it so you can get the graphs and trends you're looking for. ESPHOME is the piece that interfaces with the temp/humidity sensor and formats the data for consumption by Home Assistant. (From ESP in the name, you've probably guessed you're looking at changing the Pico W for an ESP8266 or ESP32.)

If you really, really want to do it yourself...

First, I'd suggest getting the web page you have on the Pico to display the data in JSON format. Example: `{"temp": 32, "humidity": 50}` Your web page would be behaving like and API and this will make it easier for whatever you have running on the Pi 3 to read the data. (Sorry, the link in your post didn't work for me, so I don't have any details on what you've set up.)

For the Pi, you can periodically poll the web page on your Pico to get the data. I'm assuming you're wanting to use Python to do this. If so, you'll want to look at the `requests` and `json` modules to get the data from your Pico. (I have an example of [reading an API with Python](https://github.com/DavesCodeMusings/SolarEdgeAPI) that might help you.)

As far as stuffing the info into a database, SQLite is one option that's pretty easy to get going. Unfortunately, [my example](https://github.com/DavesCodeMusings/SolarEdgeAPI/blob/main/log-output.js) of that is written using NodeJS, but it might help get you oriented to using SQLite.)

From there, you're going to need a way to retrieve the data and graph it. Something like Grafana might help with the visualization.

**Comment by scruss** [comment]
The folks at [HiveSense](https://twelvedot.com/update-on-hivesense/) (which started as a Random Hacks of Kindness [project](https://rhok.ca/projects/hivesense-the-beehive-iot-project/)) found that bees *really* don't like sensors in the hive, and anything new gets covered in propolis
